--- runsv.c	2014-08-10 13:22:35.000000000 -0500
+++ runsv-sub.c	2017-03-11 17:42:44.255741418 -0600
@@ -386,7 +386,20 @@
   int fd;
   int r;
   char buf[256];
-
+  
+  #ifdef linux
+    // We use this to see if we should use prctl() to set us as child subreaper. (And import prctl.h)
+    #define PLATFORM "linux"
+    #include <sys/prctl.h>
+  #endif
+
+  if ( PLATFORM == "linux" ) {
+	// Set ourself's as subreaper.
+	prctl(PR_SET_CHILD_SUBREAPER, 1);
+	// Ignore SIGCHLD
+	signal(SIGCHLD, SIG_IGN);
+  }
+  
   progname =argv[0];
   if (! argv[1] || argv[2]) usage();
   dir =argv[1];
@@ -546,6 +559,7 @@
 
     while (read(selfpipe[0], &ch, 1) == 1)
       ;
+
     for (;;) {
       int child;
       int wstat;
